compiler-core/src/ast.rs

```diff
@@ -696,10 +700,12 @@ pub type UntypedModuleConstant = ModuleConstant<(), ()>;
 /// pub const end_year = 2111
 /// ```
 pub struct ModuleConstant<T, ConstantRecordTag> {
-    pub documentation: Option<EcoString>,
+    pub documentation: Option<(u32, EcoString)>,
+    /// The location of the constant, starting at the "(pub) const" keywords and
+    /// ending after the ": Type" annotation, or (without an annotation) after its name.
     pub location: SrcSpan,
     pub publicity: Publicity,
-    pub name: EcoString,
+    pub name: (SrcSpan, EcoString),
```

Comment:
This should be two named fields as it's not an optional field which forces them to be grouped together.

What's the srcspan used for? I couldn't see where it is used.

Reply:
Name spans are needed for symbols. This one is used here: https://github.com/gleam-lang/gleam/blob/95e11ffd80b7ba9eaeda22c45445dd7d10535033/compiler-core/src/language_server/engine.rs#L405